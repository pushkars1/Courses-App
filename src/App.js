import React from 'react';
import CoursesTable from './components/CoursesTable';
import Header from './components/Header';

function App(props) {

  return (
    <div className="App">
      <Header />
      <CoursesTable /> 
    </div>
  );
}

export default App;
