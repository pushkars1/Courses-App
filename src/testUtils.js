import rootReducer from './reducers';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import configureMockStore from "redux-mock-store";

export const storeFactory  = (initialState={}) => {
    return createStore(rootReducer, initialState, applyMiddleware(thunk));
}

export const mockStore = configureMockStore([thunk]);
