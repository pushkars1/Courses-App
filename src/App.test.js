import {mount} from 'enzyme';
import App from './App';
import React from 'react';
import {storeFactory} from './testUtils';

describe('App component', () => {
    const store = storeFactory({});
    let app = mount(<App store={store}/>);

    test("renders without error", () => {
        const component = app.find('.App');
        expect(component.length).toBe(1);
    })

    test("snapshotTesting of App Component", () => {
        expect(app).toMatchSnapshot();
    })
})