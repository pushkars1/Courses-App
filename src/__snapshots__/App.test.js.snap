// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshotTesting of App Component 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Connect(CoursesTable) />
</div>
`;
